### PR TITLE
Get server data through JInput

### PR DESCRIPTION
**Pull request by wilsonge**
A result of the comment in https://github.com/joomla-projects/joomla-pythagoras/issues/2

My head is pounding too hard to work out the correct filtering at the moment - so just doing raw filtering.

### CODE REVIEW

**Comment by mbabker** [comment]
Isn't everything in the `$_SERVER` global that we're accessing a string?

**Comment by wilsonge** [comment]
Yah I guess so. It's just whether we can be even more restrictive than that on some the vars

**Comment by mbabker** [comment]
Considering what data we're accessing, IMO this isn't a case where you have to go most restrictive possible or raw.  Unless hacks in PHP because of spoofed server vars are really that prevalent, a string filter is really just fine here.

File: src/AbstractWebApplication.php [hunk 1]
```diff
@@ -589,21 +589,21 @@ protected function detectRequestUri()
 		 */
 
 		// If PHP_SELF and REQUEST_URI are both populated then we will assume "Apache Mode".
-		if (!empty($_SERVER['PHP_SELF']) && !empty($_SERVER['REQUEST_URI']))
+		if (!empty($this->input->server->getRaw('PHP_SELF')) && !empty($this->input->server->getRaw('REQUEST_URI')))
```
Comment (mbabker) [inline src/AbstractWebApplication.php]: You can't call `empty(method())` on PHP 5.3 or 5.4.

Comment (wilsonge) [inline src/AbstractWebApplication.php]: Fixed.

**Comment by mbabker** [comment]
We really need to refactor the app tests to inject Input objects in (or mocks) with expected test data instead of manipulating the globals...

**Comment by wilsonge** [comment]
Yup. Although as we are going to need to radically refactor JInput for PSR-7 support (injecting the data that comes out the request object I think that should fix things up then

**Pull request merged**